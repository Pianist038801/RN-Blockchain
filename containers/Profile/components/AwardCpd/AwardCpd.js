import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import NameDateBlock from '../NameDateBlock'
import VerifiedIcon from '../VerifiedIcon'
import styles from '../../../../style.js'

const AwardCpd = (props) => {
  const {
    award,
    name,
    line2
  } = props

  const dateFrom = new Date(award.date_from)
  const dateTo = new Date(award.date_to)

  return (
    <View style={styles.profileEntityContainer}>
      <VerifiedIcon type="award" entity={award} />
      <NameDateBlock
        name={name}
        line2={line2}
        dateFrom={dateFrom.toLocaleDateString('en-GB')}
        dateTo={dateTo.toLocaleDateString('en-GB')}
      />
    </View>
  )
}

AwardCpd.propTypes = {
  name: PropTypes.string.isRequired,
  line2: PropTypes.string.isRequired,
  award: PropTypes.shape({
    date_from: PropTypes.string,
    date_to: PropTypes.string
  })
}

export default AwardCpd
