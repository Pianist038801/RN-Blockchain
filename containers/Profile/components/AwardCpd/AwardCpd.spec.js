import React from 'react'
import renderer from 'react-test-renderer'
import AwardCpd from './index'

describe('containers/Profile/components/AwardCpd', () => {
  it('renders correctly', () => {
    const props = {
      name: 'hello',
      line2: 'Lorem ipsum',
      award: {
        date_from: '2017-11-04',
        date_to: '2018-01-24'
      }
    }

    const tree = renderer.create(
      <AwardCpd {...props} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
