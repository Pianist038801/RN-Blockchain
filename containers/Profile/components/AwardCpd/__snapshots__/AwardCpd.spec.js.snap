// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/AwardCpd renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#fff",
      "borderWidth": 1,
      "flex": 0.5,
      "flexDirection": "row",
      "marginBottom": 5,
      "padding": 22.5,
      "width": 600,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={1}
      style={
        Object {
          "height": 75,
          "resizeMode": "contain",
          "width": 75,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 22.5,
        "width": 360,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#569cee",
          "fontWeight": "bold",
        }
      }
    >
      hello
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Lorem ipsum
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      11/4/2017
       - 
      1/24/2018
    </Text>
  </View>
</View>
`;
